lib/WP_Auth0_Options_Generic.php

```diff
@@ -1,69 +1,221 @@
 <?php
+/**
+ * Contains WP_Auth0_Options_Generic.
+ *
+ * @package WP-Auth0
+ */
 
+/**
+ * Class WP_Auth0_Options_Generic.
+ */
 class WP_Auth0_Options_Generic {
+
+	/**
+	 * Name used in options table option_name column.
+	 *
+	 * @var string
+	 */
 	protected $_options_name = '';
-	private $_opt            = null;
 
+	/**
+	 * Current array of options stored in memory.
+	 *
+	 * @var null|array
+	 */
+	private $_opts = null;
+
+	/**
+	 * Array of options overridden by constants.
+	 *
+	 * @var array
+	 */
+	protected $constant_opts = array();
+
+	/**
+	 * WP_Auth0_Options_Generic constructor.
+	 * Finds and stores all constant-defined settings values.
+	 */
+	public function __construct() {
+		$option_keys = $this->get_defaults( true );
+		foreach ( $option_keys as $key ) {
+			if ( $this->has_constant_val( $key ) ) {
+				$this->constant_opts[ $key ] = $this->get_constant_val( $key );
+			}
+		}
+	}
+
+	/**
+	 * Takes an option key and creates the constant name to look for.
+	 *
+	 * @param string $key - Option key to transform.
+	 *
+	 * @return string
+	 */
+	public function get_constant_name( $key ) {
+		$constant_prefix = apply_filters( 'wp_auth0_settings_constant_prefix', 'AUTH0_ENV_' );
+		return $constant_prefix . strtoupper( $key );
+	}
+
+	/**
+	 * Does a certain option pull from a constant?
+	 *
+	 * @param string $key - Option key to check.
+	 *
+	 * @return boolean
+	 */
+	public function has_constant_val( $key ) {
+		$setting_const = $this->get_constant_name( $key );
+		return defined( $setting_const );
+	}
+
+	/**
+	 * Get the value of an overriding constant if one is set, return null if not.
+	 *
+	 * @param string $key - Option key to look for.
+	 *
+	 * @return string|null
+	 */
+	public function get_constant_val( $key ) {
+		$constant_name = $this->get_constant_name( $key );
+		return defined( $constant_name ) ? constant( $constant_name ) : null;
```

Comment:
it sounds like this method is repeating the logic on the `has_constant_val` method. Do you think this can be improved? i.e. if tomorrow the logic to check whether a constant exists or not changes, you'll need to change 2 methods instead of 1.